### PR TITLE
move etag/match_condition of delete into kwargs

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 1]
```diff
@@ -357,19 +357,17 @@ def set_configuration_setting(
...
         :param key: key used to identify the ConfigurationSetting
         :type key: str
         :param label: label used to identify the ConfigurationSetting
         :type label: str
-        :param etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
-        :type etag: str or None
-        :param match_condition: the match condition to use upon the etag
-        :type MatchConditions: :class:`MatchConditions`
+        :keyword str etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
+        :keyword MatchConditions match_condition: the match condition to use upon the etag
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: :keyword ~azure.core.MatchConditions match_condition:

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 2]
```diff
@@ -385,6 +383,8 @@ def delete_configuration_setting(
                 key="MyKey", label="MyLabel"
             )
         """
+        etag = kwargs.pop("etag", "*")
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: That's a change of behavior, before the default value was None. Expected?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py [hunk 3]
```diff
@@ -370,19 +370,17 @@ def list_configuration_settings(
...
         :param key: key used to identify the ConfigurationSetting
         :type key: str
         :param label: label used to identify the ConfigurationSetting
         :type label: str
-        :param etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
-        :type etag: str or None
-        :param match_condition: the match condition to use upon the etag
-        :type MatchConditions: :class:`MatchConditions`
+        :keyword str etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
+        :keyword MatchConditions match_condition: the match condition to use upon the etag
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py]: :keyword ~azure.core.MatchConditions match_condition:

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py [hunk 4]
```diff
@@ -399,6 +397,9 @@ def list_configuration_settings(
                 key="MyKey", label="MyLabel"
             )
         """
+
+        etag = kwargs.pop("etag", "*")
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py]: That's a change of behavior, before the default value was None. Expected?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 2 above)
Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: No. You are right. Bug.... Thx.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 5]
```diff
@@ -298,8 +297,7 @@ def set_configuration_setting(
         :param configuration_setting: the ConfigurationSetting to be added (if not exists) \
         or updated (if exists) to the service
         :type configuration_setting: :class:`ConfigurationSetting`
-        :param match_condition: the match condition to use upon the etag
-        :type MatchConditions: :class:`MatchConditions`
+        :param `azure.core.MatchConditions match_condition: the match condition to use upon the etag
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: not ` but ~

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 6]
```diff
@@ -357,19 +355,17 @@ def set_configuration_setting(
...
         :param key: key used to identify the ConfigurationSetting
         :type key: str
         :param label: label used to identify the ConfigurationSetting
         :type label: str
-        :param etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
-        :type etag: str or None
-        :param match_condition: the match condition to use upon the etag
-        :type MatchConditions: :class:`MatchConditions`
+        :keyword str etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
+        :keyword `azure.core.MatchConditions match_condition: the match condition to use upon the etag
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: not ` but ~

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py [hunk 7]
```diff
@@ -199,8 +199,7 @@ def list_configuration_settings(
         :type label: str
         :param etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
         :type etag: str or None
-        :param match_condition: the match condition to use upon the etag
-        :type MatchConditions: :class:`MatchConditions`
+        :param `azure.core.MatchConditions match_condition: the match condition to use upon the etag
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py]: not ` but ~

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py [hunk 8]
```diff
@@ -310,8 +309,7 @@ def list_configuration_settings(
         :param configuration_setting: the ConfigurationSetting to be added (if not exists) \
         or updated (if exists) to the service
         :type configuration_setting: :class:`ConfigurationSetting`
-        :param match_condition: the match condition to use upon the etag
-        :type MatchConditions: :class:`MatchConditions`
+        :param `azure.core.MatchConditions match_condition: the match condition to use upon the etag
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py]: not ` but ~

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py [hunk 9]
```diff
@@ -370,19 +368,17 @@ def list_configuration_settings(
...
         :param key: key used to identify the ConfigurationSetting
         :type key: str
         :param label: label used to identify the ConfigurationSetting
         :type label: str
-        :param etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
-        :type etag: str or None
-        :param match_condition: the match condition to use upon the etag
-        :type MatchConditions: :class:`MatchConditions`
+        :keyword str etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
+        :keyword `azure.core.MatchConditions match_condition: the match condition to use upon the etag
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py]: not ` but ~

File: sdk/appconfiguration/azure-appconfiguration/HISTORY.md [hunk 10]
```diff
@@ -3,6 +3,12 @@
 # Release History
 
 -------------------
+## 2019-10-xx Version 1.0.0b5
+
+### Breaking changes
+
+- delete_configuration_setting does not take etag and match_condition anymore (use kwargs instead)  #8161
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/HISTORY.md]: I would rephrase "etag and match_condition are now keyword argument only"

The way you say it "does not take [...] anymore" makes think there are not available anymore

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 11]
```diff
@@ -357,19 +355,17 @@ def set_configuration_setting(
 
     @distributed_trace
     def delete_configuration_setting(
-        self, key, label=None, etag=None, match_condition=MatchConditions.Unconditionally, **kwargs
-    ):  # type: (str, str, str, MatchConditions, dict) -> ConfigurationSetting
+        self, key, label=None, **kwargs
+    ):  # type: (str, str, dict) -> ConfigurationSetting
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py]: mypy detail: (str, Optional[str], dict)

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py [hunk 12]
```diff
@@ -370,19 +368,17 @@ def list_configuration_settings(
 
     @distributed_trace_async
     async def delete_configuration_setting(
-        self, key, label=None, etag=None, match_condition=MatchConditions.Unconditionally, **kwargs
-    ):  # type: (str, str, str, MatchConditions, dict) -> ConfigurationSetting
+        self, key, label=None, **kwargs
+    ):  # type: (str, str, dict) -> ConfigurationSetting
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py]: mypy detail: (str, Optional[str], dict)
